opengovsg/postmangovsg
# Release 1.26.0

## Improvements
- Standardise whitespace in sql statements
- Add tests for saving credentials
- Decrease memory usage in verify backup service
- Log the subtype when emails bounce/ arecomplained as spam
- Add sequelize migrations

## Deploy notes

```
ALTER TABLE email_messages
  ADD error_sub_type VARCHAR(255);

ALTER TABLE email_ops
  ADD error_sub_type VARCHAR(255);

CREATE TABLE sequelize_meta (name VARCHAR(255) PRIMARY KEY);

CREATE TABLE sequelize_data (name VARCHAR(255) PRIMARY KEY);

INSERT INTO "public"."sequelize_meta" ("name") VALUES
('20210422031603-create-users.js'),
('20210422031959-create-credentials.js'),
('20210422033007-create-telegram-subscribers.js'),
('20210422071142-create-email-blacklist.js'),
('20210422071524-create-email-from-address.js'),
('20210422072053-create-workers.js'),
('20210422072227-create-user-features.js'),
('20210422072817-create-user-demos.js'),
('20210422073013-create-campaigns.js'),
('20210422073241-create-bot-subscribers.js'),
('20210422073531-create-unsubscribers.js'),
('20210422073649-create-statistics.js'),
('20210422073931-create-sms-messages.js'),
('20210422080646-create-user-credentials.js'),
('20210422080937-create-email-templates.js'),
('20210422081139-create-sms-templates.js'),
('20210422081147-create-telegram-templates.js'),
('20210422081533-create-job-queue.js'),
('20210422081728-create-protected-messages.js'),
('20210422081912-create-email-messages.js'),
('20210422082154-create-telegram-messages.js'),
('20210422083045-create-sms-ops.js'),
('20210422083254-create-telegram-ops.js'),
('20210422083258-create-email-ops.js'),
('20210429042412-create-get-next-job-function.js'),
('20210429044423-create-insert-job-function.js'),
('20210429045222-create-is-credential-used-function.js'),
('20210501120241-create-resume-worker-function.js'),
('20210501120857-create-retry-jobs-function.js'),
('20210501121338-create-stop-jobs-function.js'),
('20210501121846-create-update-stats-email-function.js'),
('20210501125630-create-update-stats-sms-function.js'),
('20210501125840-create-update-stats-telegram-function.js'),
('20210501133459-create-enqueue-messages-email-function.js'),
('20210502051539-create-enqueue-messages-sms-function.js'),
('20210502052853-create-enqueue-messages-telegram-function.js'),
('20210502060116-create-get-messages-to-send-email-function.js'),
('20210502060418-create-get-messages-to-send-sms-function.js'),
('20210502060550-create-get-messages-to-send-telegram-function.js'),
('20210502063332-create-log-job-email-function.js'),
('20210502063415-create-log-job-sms-function.js'),
('20210502063420-create-log-job-telegram-function.js'),
('20210502063953-create-log-next-job-email-function.js'),
('20210502063956-create-log-next-job-sms-function.js'),
('20210502064000-create-log-next-job-telegram-function.js'),
('20210502064847-create-agencies-table.js');

INSERT INTO "public"."sequelize_data" ("name") VALUES
('20210422101939-add-default-credentials.js');
```

## Tests
- Confirm that subtypes are recorded
- Confirm that each type of campaign can be sent
- Confirm that sequelize-cli can be used on prod (`PGSSLMODE=no-verify npx sequelize-cli db:migrate` should say that everything is up to date)

## Todos:
- Add statement for adding subtype 
- Run sequelize migration in CI/CD
